cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/partitioned/PartitionExploreCorrector.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset.lib.partitioned;
```

Comment:
I was following the model of AbstractCubeHttpHandler, which is similar to this, a program specific to a dataset type, in c.c.c.api.dataset.lib.cube. But I agree it is better to have a separate package. Will move this, and we should move the service, too (which can be done when we address https://issues.cask.co/browse/CDAP-4159).